soc/Kconfig.v2

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+
+# SPDX-License-Identifier: Apache-2.0
+
+# Load all SoCs.
+# SoCs that has transistioned to SoC scheme v2 are promptless, and thus
+# unselectable by the user or thorugh configuration files.
+#
+# SoC that are still using scheme v1 are still having a prompt but is loaded as
+# they offer board maintainers the possibility to use board scheme v2.
+
+config SOC
+	string
+	help
+	  SoC name which can be found under soc/<arch>/<soc name>.
+	  This option holds the directory name used by the build system to locate
+	  the correct linker and header files for the SoC.
+
+config SOC_SERIES
+	string
+	help
+	  SoC series name which can be found under soc/<arch>/<family>/<series>.
+	  This option holds the directory name used by the build system to locate
+	  the correct linker and header files.
+
+config SOC_FAMILY
+	string
+	help
+	  SoC family name which can be found under soc/<arch>/<family>.
+	  This option holds the directory name used by the build system to locate
+	  the correct linker and header files.
+
+menu "Hardware Configuration"
+
+# This loads Zephyr base SoC Kconfigs for both hw model v1 and v2
+osource "soc/*/Kconfig"
+
+# Source all Kconfig HWMv2 from SoC roots.
+source "$(KCONFIG_BINARY_DIR)/Kconfig.soc"
+
+# Source Zephyr Kconfig specifics from SoC roots.
+source "$(KCONFIG_BINARY_DIR)/Kconfig.zephyr"
+
+module = SOC
+module-str = SOC
+source "subsys/logging/Kconfig.template.log_config"
+
+endmenu
+
+# The helper symbols below are put here due to an unusual setup: The simulated
+# nrf5x_bsim boards use the POSIX arch, but are compatible with Nordic ARM boards
+
+config SOC_COMPATIBLE_NRF
+	bool
+
+config SOC_COMPATIBLE_NRF52X
+	bool
+
+config SOC_COMPATIBLE_NRF53X
+	bool
+
+config SOC_COMPATIBLE_NRF52833
+	bool
+
+config SOC_COMPATIBLE_NRF5340_CPUNET
+	bool
+
+config SOC_COMPATIBLE_NRF5340_CPUAPP
+	bool
```

Comment:
Not requesting changes but wondering if a better place for this would be e.g. `Kconfig.v2.bsim_hacks` or similar

Reply:
good point.

Will consider better location.

Reply:
agree here, this is very confusing